components/script/dom/htmlimageelement.rs

```diff
@@ -1032,3 +1052,216 @@ fn image_dimension_setter(element: &Element, attr: LocalName, value: u32) {
+                        }
+                    }
+                }
+                ParseState::InParens => {
+                    match nextChar {
+                        Some((idx, c @ ')')) => {
+                            current_descriptor.push(c);
+                            state = ParseState::InDescriptor;
+                            continue;
+                        }
+                        Some((_, c)) => {
+                            current_descriptor.push(c);
+                            continue;
+                        }
+                        None => {
+                            if !current_descriptor.is_empty() {
+                                descriptors.push_back(current_descriptor.clone());
+                            }
+                            break;
+                        }
+                    }
+                }
+                ParseState::AfterDescriptor => {
+                    match nextChar {
+                        Some((idx, ' ')) => {
+                            state = ParseState::AfterDescriptor;
+                            continue;
+                        }
+                        Some((idx, c)) => {
+                            state = ParseState::InDescriptor;
+                            buffered = Some((idx, c));
+                            continue;
+                        }
+                        None => {
+                            if !current_descriptor.is_empty() {
+                                descriptors.push_back(current_descriptor.clone());
+                            }
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+        let mut error = false;
+        let mut width: Option<u32> = None;
+        let mut density: Option<f64> = None;
+        let mut future_compat_h: Option<u32> = None;
+        for descriptor in descriptors {
+            let char_iter = descriptor.chars();
+            let (digits, remaining) = collect_sequence_characters(&descriptor, is_ascii_digit);
+            let valid_non_negative_integer = parse_unsigned_integer(digits.chars());
+            let has_w = if remaining == "w" {
+                true
+            } else {
+                false
+            };
+            let valid_floating_point = parse_double(digits);
+            let has_x = if remaining == "x" {
+                true
+            } else {
+                false
+            };
+            let has_h = if remaining == "h" {
+                true
+            } else {
+                false
+            };
+            if valid_non_negative_integer.is_ok() && has_w {
+                //not support sizes attribute
+                if width.is_some() && density.is_some() {
+                    error = true;
+                }
+                let result = parse_unsigned_integer(char_iter.clone());
+                error = result.is_err();
+                if let Ok(w) = result {
```

Comment:
As you are just pulling the value out and converting it to option type (i.e, `Some(w)`), you can use the ok method like: `width = result.ok()`. Calling `ok()` on a `Result<T>` type converts it to an `Option<T>`. `ok()` method does what you are currently doing here.